packages/block-library/src/file/index.php

```diff
@@ -16,33 +16,43 @@
  * @return string Returns the block content.
  */
 function render_block_core_file( $attributes, $content ) {
+	if ( empty( $attributes['displayPreview'] ) ) {
+		return $content;
+	}
+
 	// If it's interactive, enqueue the script module and add the directives.
-	if ( ! empty( $attributes['displayPreview'] ) ) {
-		wp_enqueue_script_module( '@wordpress/block-library/file/view' );
+	wp_enqueue_script_module( '@wordpress/block-library/file/view' );
 
-		$processor = new WP_HTML_Tag_Processor( $content );
-		$processor->next_tag();
+	$processor = new WP_HTML_Tag_Processor( $content );
+	if ( $processor->next_tag() ) {
 		$processor->set_attribute( 'data-wp-interactive', 'core/file' );
-		$processor->next_tag( 'object' );
-		$processor->set_attribute( 'data-wp-bind--hidden', '!state.hasPdfPreview' );
-		$processor->set_attribute( 'hidden', true );
-
-		$filename     = $processor->get_attribute( 'aria-label' );
-		$has_filename = ! empty( $filename ) && 'PDF embed' !== $filename;
-		$label        = $has_filename ? sprintf(
-			/* translators: %s: filename. */
-			__( 'Embed of %s.' ),
-			$filename
-		) : __( 'PDF embed' );
-
-		// Update object's aria-label attribute if present in block HTML.
-		// Match an aria-label attribute from an object tag.
-		$processor->set_attribute( 'aria-label', $label );
-
-		return $processor->get_updated_html();
 	}
 
-	return $content;
+	while ( 'OBJECT' !== $processor->get_tag() && $processor->next_tag( 'object' ) ) {
+		continue;
+	}
```

Comment:
this and the lines down to 38 below should be equivalent to the following…

```php
// If there are no OBJECT elements, something might have already modified the block.
if ( ! $processor->next_tag( 'OBJECT' ) ) {
	return $content;
}
```

if I’m understanding correctly, the goal is to find the first `OBJECT` element, and return the `$content` if none is found.

the `while` is going to stop at the first `OBJECT` opening tag because once it finds one, `'OBJECT' !== $processor->get_tag()` will be false unless it failed to find one at all, which is picked up below.

Reply:
Thanks, @dmsnell!

That makes sense and is more convenient.